kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -534,43 +429,29 @@ protected Optional<LogSegment> getLogSegmentForVersion(
       Engine engine,
       Optional<Long> startCheckpointOpt,
       Optional<Long> versionToLoadOpt,
-      Optional<List<FileStatus>> filesOpt) {
-    final List<FileStatus> newFiles;
-    if (filesOpt.isPresent()) {
-      newFiles = filesOpt.get();
-    } else {
-      // No files found even when listing from 0 => empty directory =>
-      // table does not exist yet.
-      if (!startCheckpointOpt.isPresent()) {
-        return Optional.empty();
+      List<FileStatus> newFiles) {
+    if (newFiles.isEmpty()) {
+      if (startCheckpointOpt.isPresent()) {
+        // We either (a) determined this checkpoint version from the _LAST_CHECKPOINT file, or (b)
+        // found the last complete checkpoint before our versionToLoad. In either case, we didn't
+        // see the checkpoint file in the listing.
+        // TODO: for case (a), re-load the delta log but ignore the _LAST_CHECKPOINT file.
```

Comment:
Sorry can you elaborate on case (b)? I thought `startCheckpointOpt` was only populated from the _last_checkpoint file